src/components/plugin-selector-dialog.js

```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
+import { Modal } from 'antd';
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import InfoFactory from '../plugins/info-factory.js';
+import { useService } from './container-context.js';
+import GridSelector from './grid-selector.js';
+
+function PluginSelectorDialog({ visible, onSelected, onCancel }) {
+  const { t } = useTranslation('pluginSelectorDialog');
+  const infoFactory = useService(InfoFactory);
+
+  const pluginItems = useMemo(() => {
+    return infoFactory.getRegisteredTypes()
+      .map(typeName => infoFactory.createInfo(typeName))
+      .map(info => ({
+        key: info.type,
+        icon: info.getIcon?.(t) || null,
```

Comment:
We could use a default icon here, maybe `<QuestionOutlined />`

Reply:
Good idea, changed, thanks!